bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Rule.java

```diff
@@ -146,7 +156,7 @@ public void setName(String ruleName) throws IllegalStateException {
      * @return a {@link Set} of tags
      */
     public Set<String> getTags() {
-        return tags = tags != null ? tags : Collections.<String> emptySet();
+        return tags;
```

Comment:
Do you return the internal set by intention, so it could be modified by the caller?
Shouldn't we return a copy of the current set, so a modification does not result in a changed tag set without calling setTag.
Another point is if two consumers call getTags() and one iterate over the returned reference while the other one modifies the returned reference the first one could run into a CME.

Also the set function should create a new set with the content of the supplied argument instead of using the given reference that could be changed by the caller and so the tags could be modified without calling setTags.

Reply:
for the user convenience, this will be fixed in RuleUtils class